### ISSUE TITLE
NPE in RemoteBufferFileImpl.dispose

### ISSUE THREAD

**Issue by mumbel** [NationalSecurityAgency/ghidra#3962]
**Describe the bug**
Ghidra server was killed from an OOM issue looks like, which is maybe just VM/JVM/config related.

But when looking at the log I noticed there were a lot of NPE at:
https://github.com/NationalSecurityAgency/ghidra/blob/da94eb86bd2b89c8b0ab9bd89e9f0dc5a3157055/Ghidra/Features/GhidraServer/src/main/java/db/buffers/RemoteBufferFileImpl.java#L211

and the only other Ghidra loc of in trace was:
https://github.com/NationalSecurityAgency/ghidra/blob/da94eb86bd2b89c8b0ab9bd89e9f0dc5a3157055/Ghidra/Features/GhidraServer/src/main/java/db/buffers/RemoteBufferFileImpl.java#L171

The rest of the trace was (I can copy the real trace if needed next week):
Target.lambda$refSetRemove$0
AccessController.doPrivileged
Target.lambda$refSetRemove$1
Thread.run

**Environment (please complete the following information):**
 - OS: 16.04
 - Java Version: 11.0.2
 - Ghidra Version: 10.1
 - Ghidra Origin: github

**Comment by ghidra1** [comment]
The real stack trace would be very useful.  The disposal is not thread-safe and does not handle multi-threaded use.  I will need to evaluate further based on the stack traces you are able to provide.  Thanks.

**Comment by mumbel** [comment]
```
FINEST|4839/0|22-02-04 00:14:12|INFO  [FOOBAR]: handle disposed (mumbel) (RepositoryManager)  
FINEST|4839/0|22-02-04 00:14:12|Uncaught exception by Thread[RMI Unreferenced-148,5,RMI Runtime]:
FINEST|4839/0|22-02-04 00:14:12|java.lang.NullPointerException:null
FINEST|4839/0|22-02-04 00:14:12|java.lang.NullPointerException
FINEST|4839/0|22-02-04 00:14:12|	at db.buffers.RemoteBufferFileImpl.dispose(RemoteBufferFileImpl.java:211)
FINEST|4839/0|22-02-04 00:14:12|	at db.buffers.RemoteBufferFileImpl.unreferenced(RemoteBufferFileImpl.java:171)
FINEST|4839/0|22-02-04 00:14:12|	at java.rmi/sun.rmi.transport.Target.lambda$refSetRemove$0(Target.java:330)
FINEST|4839/0|22-02-04 00:14:12|	at java.base/java.security.AccessController.doPrivileged(Native Method)
FINEST|4839/0|22-02-04 00:14:12|	at java.rmi/sun.rmi.transport.Target.lambda$refSetRemove$1(Target.java:329)
FINEST|4839/0|22-02-04 00:14:12|	at java.base/java.lang.Thread.run(Thread.java:834)
```

@ghidra1   I think most, not all, looked like this and the `handle disposed` message was in the same tick.

**Comment by ghidra1** [comment]
The use of buffer files and related database API's are generally guarded by synchronization at a higher level.   The spurious non-synchronized call to unreferenced has no protection.  I can look into protecting the dispose method with some synchronization.

**Comment by ghidra1** [comment]
Fix is in the works for 10.2
